Code/GraphMol/RGroupDecomposition/RGroupScore.cpp

```diff
@@ -110,13 +110,14 @@ double matchScore(const std::vector<size_t> &permutation,
     }
 
     // overweight linkers with the same attachments points....
-    //  because these belong to 2 rgroups we really want these to stay
+    //  because these belong to 2 (or more) rgroups we really want these to stay
+    //  the size of the set is the number of labels that are being used
     //  ** this heuristic really should be taken care of above **
     int maxLinkerMatches = 0;
     for (const auto &it : linkerMatchSet) {
-      if (it.second > 1) {
-        if (it.second > maxLinkerMatches) {
-          maxLinkerMatches = it.second;
+      if (it.first.size() > 1 || it.second > 1) {
```

Comment:
first.size is the set of r-labels used, it.second is the number of times this r-groups is attached.  This second test has been obsoleted somehow, it used to be that there was only one rgroup recordered per label set, but now they are independent so this won't get above one.